detection_rules/packaging.py

```diff
@@ -74,77 +74,56 @@ def load_versions(current_versions: dict = None):
     return current_versions or load_etc_dump('version.lock.json')
 
 
-def manage_versions(rules: List[TOMLRule], deprecated_rules: list = None, current_versions: dict = None,
+def manage_versions(rules: List[TOMLRule], current_versions: dict = None,
                     exclude_version_update=False, add_new=True, save_changes=False,
                     verbose=True) -> (List[str], List[str], List[str]):
     """Update the contents of the version.lock file and optionally save changes."""
     current_versions = load_versions(current_versions)
-    new_versions = {}
+    versions_hash = dict_hash(current_versions)
+    rule_deprecations = load_etc_dump('deprecated_rules.json')
 
-    for rule in rules:
-        new_versions[rule.id] = {
-            'sha256': rule.contents.sha256(),
-            'rule_name': rule.name,
-            'version': rule.contents.autobumped_version
-        }
+    echo = click.echo if verbose else (lambda x: None)
 
-    new_rules = [rule for rule in rules if rule.contents.latest_version is None]
-    changed_rules = [rule for rule in rules if rule.contents.is_dirty]
-    # manage deprecated rules
-    newly_deprecated = []
-    rule_deprecations = {}
+    already_deprecated = set(rule_deprecations)
+    deprecated_rules = set(rule.id for rule in rules if rule.contents.metadata.maturity == "deprecated")
+    new_rules = set(rule.id for rule in rules if rule.contents.latest_version is None) - deprecated_rules
```

Comment:
if a rule was never version, it shouldn't need to be deprecated, just removed. At least that was normally the case, but with the way we have been versioning _after_ that may not always be true

Reply:
yeah this should never happen unless we deprecate before locking... whatever that means